tests/integration_tests/saas/test_stripe_task.py

```diff
@@ -627,3 +625,596 @@ def test_stripe_access_request_task(
+        f"{dataset_name}:subscription": 0,
+        f"{dataset_name}:dispute": 0,
+    }
+
+    stripe_secrets = stripe_connection_config.secrets
+    base_url = f"https://{stripe_secrets['host']}"
+
+    # customer
+    response = requests.get(
+        url=f"{base_url}/v1/customer", params={"email": stripe_erasure_identity_email}
+    )
+    customer = response.json()["data"][0]
+    customer_id = customer["id"]
+    assert customer["shipping"]["name"] == "MASKED"
+
+    # card
+    response = requests.get(
+        url=f"{base_url}/v1/customers/{customer_id}/sources",
+        params={"object": "card"},
+    )
+    card = response.json()
+    assert card["name"] == "MASKED"
+
+    # payment method
+    response = requests.get(
+        url=f"{base_url}/v1/customers/{customer_id}/payment_methods",
+        params={"type": "card"},
+    )
+    payment_methods = response.json()["data"]
+    for payment_method in payment_methods:
+        assert payment_method["billing_details"]["name"] == "MASKED"
+
+    # bank account
+    response = requests.get(
+        url=f"{base_url}/v1/customers/{customer_id}/sources",
+        params={"object": "bank_account"},
+    )
+    bank_account = response.json()
+    assert bank_account["account_holder_name"] == "MASKED"
+
+    # run erasure without MASKING_STRICT to execute the delete actions
+
+    config.execution.MASKING_STRICT = False
+
+    x = graph_task.run_erasure(
+        privacy_request,
+        erasure_policy_string_rewrite,
+        graph,
+        [stripe_connection_config],
+        {"email": stripe_erasure_identity_email},
+        get_cached_data_for_erasures(privacy_request.id),
+    )
+
+    # verify masking request was issued for endpoints with an update or delete action
+    assert x == {
+        f"{dataset_name}:customer": 1,
+        f"{dataset_name}:tax_id": 1,
+        f"{dataset_name}:invoice_item": 1,
+        f"{dataset_name}:charge": 0,
+        f"{dataset_name}:invoice": 2,
+        f"{dataset_name}:card": 1,
+        f"{dataset_name}:customer_balance_transaction": 0,
+        f"{dataset_name}:payment_intent": 0,
+        f"{dataset_name}:payment_method": 2,
+        f"{dataset_name}:credit_note": 0,
+        f"{dataset_name}:bank_account": 1,
+        f"{dataset_name}:subscription": 1,
+        f"{dataset_name}:dispute": 0,
+    }
+
+    # customer
+    response = requests.get(
+        url=f"{base_url}/v1/customer", params={"email": stripe_erasure_identity_email}
+    )
+    customer = response.json()["data"]
+    assert customer == []
+
+    # tax_id
+    response = requests.get(url=f"{base_url}/v1/customer/{customer_id}/tax_ids")
+    tax_ids = response.json()
+    assert tax_ids["error"]["message"] == f"No such customer: '{customer_id}'"
+
+    # invoice_item
+    response = requests.get(
+        url=f"{base_url}/v1/invoiceitems", params={"customer": {customer_id}}
+    )
+    invoice_item = response.json()["data"]
+    assert len(invoice_item) == 1
```

Comment:
Can you add a code comment explaining why there's one invoice left?  I assume it's a certain type of invoice that you're not allowed to mask/delete

Reply:
```Can't delete an invoice item that is attached to an invoice that is no longer editable.```